### PR TITLE
#1 light-rpc bootstrap

### PR DESCRIPTION
**Pull request by aniketfuryrocks**
#### Problem

#1 

#### Summary of Changes

Bootstrapped the `light-rpc` crate

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -115,9 +115,9 @@ dependencies = [
 
 [[package]]
 name = "anyhow"
-version = "1.0.58"
```
Comment (godmodegalactus) [inline Cargo.lock:118]: Any particular reason for version change.

File: light-rpc/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Light RPC
+
+Submitting a [transaction](https://docs.solana.com/terminology#transaction) to be executed on the solana blockchain,
+requires the client to identify the next few leaders based on the
+[leader schedule](https://en.wikipedia.org/wiki/QUIC), look up their peering information in gossip and
```
Comment (godmodegalactus) [inline light-rpc/README.md:5]: Leader shedule is pointing to QUIC

File: light-rpc/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+run full [validators](https://docs.solana.com/terminology#validator) that forward the transactions according to the
+protocol on behalf of the web browser. Running full solana [validators](https://docs.solana.com/terminology#validator)
+is incredibly resource intensive `(>256GB RAM)`, the goal of this
+project would be to create a specialized micro-service that allows
+to deploy this logic quickly and allows [horizontal scalability](https://en.wikipedia.org/wiki/Scalability) with
+commodity vms.
+
+## Run
+
+*make sure `solana-test-validator` is running in the background*
```
Comment (godmodegalactus) [inline light-rpc/README.md:17]: This is for testing only may be should add a new section.

File: light-rpc/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+use std::sync::Arc;
+
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_client::nonblocking::tpu_client::TpuClient;
+
+const RPC_ADDR: &str = "http://localhost:8899";
+const WEBSOCKET_ADDR: &str = "ws://localhost:8900";
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
```
Comment (godmodegalactus) [inline light-rpc/src/main.rs:10]: you can use the same `Result` definition that is used by other solana projects.

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 1 above)
Comment (aniketfuryrocks) [reply Cargo.lock:118]: I did `cargo add anyhow`. Didn't notice change. Should this be reverted ?

File: light-rpc/README.md (hunk 2 above)
Comment (aniketfuryrocks) [reply light-rpc/README.md:5]: ahh fixing

File: light-rpc/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "light-rpc"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+solana-client = { path = "../client" }
+tokio = { version = "1.14.1", features = ["full"]} 
+anyhow = "1.0.66"
```
[the remark this answers is not in the document]
Comment (aniketfuryrocks) [reply light-rpc/Cargo.toml:9]: makes error's easier in rust. Can remove it if u want.

File: light-rpc/src/main.rs (hunk 4 above)
Comment (aniketfuryrocks) [reply light-rpc/src/main.rs:10]: ohk, removing `anyhow` and replacing it with the repo wide `Result`

File: light-rpc/README.md (hunk 2 above)
Comment (aniketfuryrocks) [reply light-rpc/README.md]: fixed in [41a1a61](https://github.com/blockworks-foundation/solana/pull/7/commits/41a1a61a561420d6d32241b65e3a14150d3decdd)

File: light-rpc/src/main.rs (hunk 4 above)
Comment (aniketfuryrocks) [reply light-rpc/src/main.rs]: https://github.com/blockworks-foundation/solana/pull/7/commits/1e6148a765c34ffa65ca2dbeaab4cb65e7fedf5f

File: light-rpc/Cargo.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (aniketfuryrocks) [reply light-rpc/Cargo.toml]: https://github.com/blockworks-foundation/solana/pull/7/commits/1e6148a765c34ffa65ca2dbeaab4cb65e7fedf5f

File: light-rpc/README.md (hunk 3 above)
Comment (aniketfuryrocks) [reply light-rpc/README.md:17]: added a `##Test` section

**Review by godmodegalactus — APPROVED** [review]

**Pull request merged**